### PR TITLE
NP-51263 Format citation author list

### PR DESCRIPTION
**Pull request by iBuzza**
# Description

Link to Jira issue: https://sikt.atlassian.net/browse/NP-51263

Citation formatting now includes proper editor information with APA-style role suffixes (Ed./Eds.)
Enhanced handling of author and editor names in citations to resolve edge cases
Support MathJax for registration title

# How to test

Make sure the citations render properly for different types.

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/pages/public_registration/details-panel/_components/citation-box/CitationBox.tsx [hunk 1]
```diff
@@ -19,18 +22,30 @@ export const CitationBox = ({ registration }: CitationBoxProps) => {
...
+  const editors = parentBook ? formatAuthorList(getEditors(parentBook), { role: 'editor' }) : '';
+
+  const citation = formatAPA(registration, { journalName, publisherName, bookTitle, editors });
 
-  const citation = formatAPA(registration, { journalName, publisherName, bookTitle });
+  useEffect(() => {
+    if (stringIncludesMathJax(mainTitle) || stringIncludesMathJax(bookTitle)) {
+      typesetMathJax();
+    }
+  }, [mainTitle, bookTitle]);
```
Comment (iBuzza) [inline src/pages/public_registration/details-panel/_components/citation-box/CitationBox.tsx:49]: this `useEffect` is duplicated in several other omponents. Will create a hook for it in a later PR

File: src/pages/public_registration/details-panel/_components/citation-box/CitationBox.tsx [hunk 2]
```diff
@@ -41,17 +56,14 @@ export const CitationBox = ({ registration }: CitationBoxProps) => {
...
-        }}
-      />
+      <Paper
+        variant="outlined"
+        role="region"
+        tabIndex={0}
+        aria-labelledby={citationHeadingId}
+        sx={{ p: '1rem', maxHeight: '12rem', overflow: 'auto', whiteSpace: 'pre-wrap' }}>
+        <Typography>{citation}</Typography>
+      </Paper>
```
Comment (iBuzza) [inline src/pages/public_registration/details-panel/_components/citation-box/CitationBox.tsx:66]: Changed to `<Paper>` to allow MathJaX rendering

File: src/pages/public_registration/details-panel/_components/citation-box/_utils/formatters/format-chapter.ts
Comment (denektenina) [inline src/pages/public_registration/details-panel/_components/citation-box/_utils/formatters/format-chapter.ts:1]: Hvorfor tok du bort kommentaren? Jeg syntes den var nyttig.

File: src/pages/public_registration/details-panel/_components/citation-box/_utils/citation-helpers.ts
Comment (denektenina) [inline src/pages/public_registration/details-panel/_components/citation-box/_utils/citation-helpers.ts:1]: Det er virkelig gull at du skriver kommentarer for å forklare logikken, for de som ikke kjenner APA-reglene

File: src/pages/public_registration/details-panel/_components/citation-box/_components/CopyCitationButton.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ export const CopyCitationButton = ({ citation }: CopyCitationButtonProps) => {
     <>
       <Button
         data-testid={dataTestId.registrationLandingPage.detailsTab.copyCitationButton}
-        color={justCopied ? 'success' : 'secondary'}
+        color="secondary"
```
Comment (joachimjorgensen) [inline src/pages/public_registration/details-panel/_components/citation-box/_components/CopyCitationButton.tsx:39]: Nice

**Review by denektenina — APPROVED** [review]

**Review by joachimjorgensen — APPROVED** [review]

File: src/pages/public_registration/details-panel/_components/citation-box/_utils/formatters/format-chapter.ts
Comment (iBuzza) [reply src/pages/public_registration/details-panel/_components/citation-box/_utils/formatters/format-chapter.ts:1]: Tenkte den var overflødig siden den forklarer hva som skjer en annen plass i koden 🤔
